contrib/aws/internal/eventbridge/eventbridge.go

```diff
@@ -0,0 +1,104 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package eventbridge
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/eventbridge"
+	"github.com/aws/aws-sdk-go-v2/service/eventbridge/types"
+	"github.com/aws/smithy-go/middleware"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
+	"strconv"
+	"time"
+)
+
+const (
+	datadogKey      = "_datadog"
+	startTimeKey    = "x-datadog-start-time"
+	resourceNameKey = "x-datadog-resource-name"
+	maxSizeBytes    = 256 * 1024 // 256 KB
+)
+
+func EnrichOperation(ctx context.Context, in middleware.InitializeInput, operation string) {
+	switch operation {
+	case "PutEvents":
+		handlePutEvents(ctx, in)
+	}
+}
+
+func handlePutEvents(ctx context.Context, in middleware.InitializeInput) {
+	params, ok := in.Parameters.(*eventbridge.PutEventsInput)
+	if !ok {
+		log.Debug("Unable to read PutEvents params")
+		return
+	}
+
+	for i := range params.Entries {
+		injectTraceContext(ctx, &params.Entries[i])
+	}
+}
+
+func injectTraceContext(ctx context.Context, entryPtr *types.PutEventsRequestEntry) {
+	if entryPtr == nil {
+		return
+	}
+
+	span, ok := tracer.SpanFromContext(ctx)
+	if !ok || span == nil {
+		log.Debug("Unable to find span from context")
+		return
+	}
+
+	carrier := tracer.TextMapCarrier{}
+	err := tracer.Inject(span.Context(), carrier)
+	if err != nil {
+		log.Debug("Unable to inject trace context: %s", err)
+		return
+	}
+
+	// Add start time and resource name
+	startTimeMillis := time.Now().UnixMilli()
+	carrier[startTimeKey] = strconv.FormatInt(startTimeMillis, 10)
+	if entryPtr.EventBusName != nil {
+		carrier[resourceNameKey] = *entryPtr.EventBusName
+	}
+
+	var detail map[string]interface{}
+	if entryPtr.Detail != nil {
+		err = json.Unmarshal([]byte(*entryPtr.Detail), &detail)
+		if err != nil {
+			log.Debug("Unable to unmarshal event detail: %s", err)
+			return
+		}
+	} else {
+		detail = make(map[string]interface{})
+	}
+
+	jsonBytes, err := json.Marshal(carrier)
+	if err != nil {
+		log.Debug("Unable to marshal trace context: %s", err)
+		return
+	}
+
+	detail[datadogKey] = json.RawMessage(jsonBytes)
+
+	updatedDetail, err := json.Marshal(detail)
+	if err != nil {
+		log.Debug("Unable to marshal modified event detail: %s", err)
+		return
+	}
+
+	// Check new detail size
+	if len(updatedDetail) > maxSizeBytes {
+		log.Info("Payload size too large to pass context")
+		return
+	}
```

Comment:
There should be a way we can perform this max size test before having to do any of the detail unmarshalling and carrier injecting. It'd be best to avoid doing all of that work (especially the unmarshal/marshal cuz it's super slow) if we don't have to.

Reply:
Good idea. I did improve the efficiency by checking the size before marshaling the final updated detail.

However, we can't estimate the carrier size ahead of time. We still have to create the carrier first, since the carrier size can differ drastically depending on what DD_TRACE_PROPAGATION_STYLE is set (technically the user can pick multiple propagation styles). 

After creating the carrier, there's no easy way to get the byte size of the carrier before marshaling it to json bytes. Unless I'm missing something, the only way I can think of is:
```
size := 0
for k, v := range carrier {
	size += len(k) + len(v)
}
```

But this is an unnecessary operation in the cases that the carrier size is less than the limit.

So currently I'm doing the detail size check after the carrier is marshaled but before updated detail is marshaled.